### PR TITLE
[v6r15] More options for the VOMS2CSAgent

### PR DESCRIPTION
**Pull request by atsareg**
Added more options for the VOMS2CSAgent:

- Possibility to automatically delete users who are no longer in the VOMS ( AutoDeleteUsers CS flag )
- Possibility to create user home directories in DFC ( MakeHomeDirectory CS flag )
- Possibility to generate more detailed report for the admins ( DetailedReport CS flag )

Plus several fixes in the logic of the agent.

### CODE REVIEW

**Comment by andresailer** [comment]
Don't forget to update the documentation: http://dirac.readthedocs.io/en/latest/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents/VOMS2CSAgent/index.html

File: ConfigurationSystem/Agent/VOMS2CSAgent.py [hunk 1]
```diff
@@ -75,37 +83,143 @@ def execute( self ):
...
+
+    diracVOMSMapping = result['Value']['DIRACVOMS']
+
+    records = []
+    groupDict = {}
+    multiDNUsers = {}
+    suspendedUsers = []
+    for user in userDict:
+      for group in userDict[user]['Groups']:
+        groupDict.setdefault( group, 0 )
```
Comment (andresailer) [inline ConfigurationSystem/Agent/VOMS2CSAgent.py]: One could also use `from collections import defaultdict` for groupDict
https://docs.python.org/2/library/collections.html#collections.defaultdict

**Comment by closier** [comment]
It seems that the options AutoAddVOMS is not take into account. which means that any group where a role is defined is filled even if the option AutoAddVOMS=False

**Comment by closier** [comment]
It seems that the case where a "user" is defined in DIRAC CS but is not present in VOMS is not take into account. so if the VO is VOMS aware, we should check in both direction (VOMS - Cs but also CS - VOMS)

**Comment by atsareg** [comment]
AudoAddVOMS is an option that can be defined for a group which defines if VOMS extension should be added to a proxy generated with dirac-proxy-init ( or equivalent ) command. This is not related to the work of the VOMS2CSAgent. If you want some DIRAC groups mapped on some VOMS roles not to be updated with users having this role, this is to be discussed whether this is a real requirement.

**Comment by chaen** [comment]
Pelase add the [WIP] flag if you are still working on it, so we know we should not review yet

**Comment by atsareg** [comment]
The module is ready for review. Actually, last modifs are responses to already made remarks and also a result of testing against the LHCb case

**Pull request merged**